libpod/container_internal.go

```diff
@@ -1185,7 +1182,9 @@ func (c *Container) generateSpec(ctx context.Context) (*spec.Spec, error) {
 	}
 
 	var err error
-	if os.Getuid() == 0 {
+	if rootless.IsRootless() {
+		g.Spec().Linux.Resources = &spec.LinuxResources{}
```

Comment:
Should do this in pkg/spec instead of `libpod` - `libpod` will faithfully use whatever spec the user passes in via the API, even if the runtime will choke because it's not at all valid.

Reply:
I've dropped this part as it is not actually required.  We already have that logic in `pkg/spec`